web/client/components/share/SharePanel.jsx

```diff
@@ -151,6 +151,13 @@ class SharePanel extends React.Component {
             !isEqual(this.props.isVisible, newProps.isVisible)) {
             this.initializeDefaults(newProps);
         }
+        if (this.props?.viewerOptions?.orientation) {
```

Comment:
I think we can move this block inside `initializeDefaults` by adding a condition to check `!isEqual(this.props?.viewerOptions?.orientation, newProps.viewerOptions?.orientation)` invoking `initialzeDefaults` func

Reply:
well noted. Thanks

Reply:
@dsuren1 I think I might have to revert to the original implementation. After refactoring to this, the inputted values go beyond the specified range in certain cases.